src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -2072,5 +2131,428 @@ export class NetworkAreaDiagramViewer {
+
+        const bendLinesButton = DiagramUtils.getBendLinesButton();
+        buttonsDiv.appendChild(bendLinesButton);
+        bendLinesButton.addEventListener('click', () => {
+            if (this.bendLines) {
+                this.disableLineBending();
+                bendLinesButton.classList.remove('bend-lines-button-active');
+                bendLinesButton.classList.add('bend-lines-button-inactive');
+                bendLinesButton.title = 'Enable line bending';
+            } else {
+                this.enableLineBending();
+                if (this.bendLines) {
+                    bendLinesButton.classList.remove('bend-lines-button-inactive');
+                    bendLinesButton.classList.add('bend-lines-button-active');
+                    bendLinesButton.title = 'Disable line bending';
+                }
+            }
+        });
+        return buttonsDiv;
+    }
+
+    private enableLineBending() {
+        const linesPointsElement = document.createElementNS('http://www.w3.org/2000/svg', 'g');
+        linesPointsElement.id = 'lines-points';
+        linesPointsElement.classList.add('nad-line-points');
+        const bendableEdges = DiagramUtils.getBendableLines(this.diagramMetadata?.edges);
+        for (const edge of bendableEdges) {
+            if (edge.points) {
+                for (let index = 0; index < edge.points.length; index++) {
+                    this.addLinePoint(
+                        edge.svgId,
+                        index,
+                        new Point(edge.points[index].x, edge.points[index].y),
+                        linesPointsElement
+                    );
+                }
+                this.bendableLines.push(edge.svgId);
+            } else {
+                this.bendableLines.push(edge.svgId);
+                this.bendLines = true;
+            }
+        }
+        if (this.bendLines) {
+            this.svgDraw?.node.firstElementChild?.appendChild(linesPointsElement);
+        }
+    }
+
+    private addLinePoint(
+        lineId: string,
+        index: number,
+        point: Point,
+        linePointsElement?: SVGElement | null
+    ): SVGElement {
+        linePointsElement ??= this.svgDraw?.node.querySelector('#lines-points');
+        const pointElement = DiagramUtils.createLinePointElement(
+            lineId,
+            point,
+            index,
+            false,
+            this.linePointIndexMap,
+            this.linePointByEdgeIndexMap
+        );
+        linePointsElement?.appendChild(pointElement);
+        return pointElement;
+    }
+
+    private disableLineBending() {
+        const linePointsElement = this.svgDraw?.node.querySelector('#lines-points');
+        linePointsElement?.remove();
+        this.linePointIndexMap.clear();
+        this.linePointByEdgeIndexMap.clear();
+        this.bendableLines = [];
+        this.bendLines = false;
+    }
+
+    private moveLinePoint(svgId: string, newPosition: Point) {
```

Comment:
This method needs to be removed as it's only called when the point are undefined, which should correspond to an empty map